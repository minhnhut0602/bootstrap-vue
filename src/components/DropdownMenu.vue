<template>
    <ul class="dropdown-menu">
        <li
            v-for="(item, index) of items"
            :key="`item-${index}`"
        >
            <hr
                v-if="item.divider"
                class="dropdown-divider"
            >

            <h6
                v-else-if="item.header"
                class="dropdown-header"
            >
                {{ item.label }}
            </h6>

            <a
                v-else
                class="dropdown-item"
                :class="{active: item.active}"
                href="#"
                @click.prevent="clickItem(item)"
            >
                {{ item.label }}
            </a>
        </li>
    </ul>
</template>

<script lang="ts" setup>
import {Item, dropdownItemsProps} from '@/composables/useDropdownItems';
import {defineEmits, defineProps} from 'vue';

defineProps(dropdownItemsProps);

const emit = defineEmits(['click:item']);

const clickItem = (item: Item) => emit('click:item', item);
</script>
