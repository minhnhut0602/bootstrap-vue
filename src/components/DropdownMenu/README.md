# DropdownMenu

```vue live
<DropdownMenu>Default Example Usage</DropdownMenu>
```

## Slots

| Name    | Description | Bindings |
| ------- | ----------- | -------- |
| prepend |             |          |
| items   |             | <br/>    |
| item    |             | <br/>    |
| append  |             |          |

## Events

| Event name | Properties | Description |
| ---------- | ---------- | ----------- |
| click:item |            |
