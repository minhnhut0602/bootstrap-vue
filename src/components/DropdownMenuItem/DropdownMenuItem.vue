<template>
    <li>
        <hr
            v-if="divider"
            class="dropdown-divider"
            :class="itemClass"
        >

        <h6
            v-else-if="header"
            class="dropdown-header"
            :class="itemClass"
        >
            <!-- @slot Displays content inside the dropdown menu item. -->
            <slot name="default">
                {{ label }}
            </slot>
        </h6>

        <Component
            v-else
            :is="is"
            class="dropdown-item"
            :class="[{active: active}, itemClass]"
            :href="href"
            :to="to"
            @click.prevent
        >
            <slot name="default">
                {{ label }}
            </slot>
        </Component>
    </li>
</template>

<script lang="ts" setup>
import {dropdownItemProps} from '@/composables/useDropdownItem';
import { PropType, computed } from 'vue';

const props = defineProps({
    /**
     * Adds the `active` class
     */
    active: {
        type: Boolean,
        default: false,
    },
    /**
     * Displays a divider
     */
    divider: {
        type: Boolean,
        default: false,
    },
    ...dropdownItemProps,
    /**
     * Displays label as header
     */
    header: {
        type: Boolean,
        default: false,
    },
    href: {
        type: String,
        default: '#',
    },
    is: {
        type: String,
        default: undefined,
    },
    /**
     * The label to display
     */
    label: {
        type: String,
        default: undefined,
    },
    to: {
        type: [Object, String] as PropType<string|Record<string, unknown>>,
        default: undefined,
    },
});

const is = computed(() => {
    if (props.is) {
        return props.is;
    }

    if (props.to) {
        return 'RouterLink';
    }

    return 'a';
});
</script>

<docs>
```vue
<DropdownMenuItem label="Menu item" />
```
</docs>
