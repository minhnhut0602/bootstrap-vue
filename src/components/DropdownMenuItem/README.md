# DropdownMenuItem

```vue live
<DropdownMenuItem />
```

## Props

| Prop name | Description | Type    | Values | Default   |
| --------- | ----------- | ------- | ------ | --------- |
| active    |             | boolean | -      | false     |
| divider   |             | boolean | -      | false     |
| header    |             | boolean | -      | false     |
| label     |             | string  | -      | undefined |
