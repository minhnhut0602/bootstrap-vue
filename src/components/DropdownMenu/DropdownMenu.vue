<template>
    <ul class="dropdown-menu">
        <DropdownMenuItem
            v-for="(item, index) of items"
            :key="`item-${index}`"
            v-bind="item"
            @click="clickItem(item)"
        />
    </ul>
</template>

<script lang="ts">
import {Item, dropdownItemsProps} from '@/composables/useDropdownItems';
import DropdownMenuItem from '@/components/DropdownMenuItem';
</script>

<script lang="ts" setup>
defineProps(dropdownItemsProps);

const emit = defineEmits(['click:item']);

const clickItem = (item: Item) => emit('click:item', item);
</script>
