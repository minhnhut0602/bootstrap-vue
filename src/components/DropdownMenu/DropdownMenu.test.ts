import DropdownMenu from '../DropdownMenu';
import DropdownMenuItem from '../DropdownMenuItem';
import {shallowMount} from '@vue/test-utils';

const items = [
    {header: true, label: 'Item 1'},
    {label: 'Item 3'},
    {divider: true},
    {label: 'Item 4', active: true},
];

describe('template', () => {
    componentRenderTest(DropdownMenu, {props: {items}});

    it('emits event on click dropdown menu item', async() => {
        const wrapper = shallowMount(DropdownMenu, {props: {items}});

        await wrapper.findComponent(DropdownMenuItem).trigger('click');

        expect(wrapper.emitted('click:item')[0]).toEqual([items[0]]);
    });
});
