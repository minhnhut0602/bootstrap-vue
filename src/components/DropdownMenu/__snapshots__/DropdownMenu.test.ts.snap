// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<ul class=\\"dropdown-menu\\">
  <dropdown-menu-item-stub active=\\"false\\" divider=\\"false\\" header=\\"true\\" label=\\"Item 1\\"></dropdown-menu-item-stub>
  <dropdown-menu-item-stub active=\\"false\\" divider=\\"false\\" header=\\"false\\" label=\\"Item 3\\"></dropdown-menu-item-stub>
  <dropdown-menu-item-stub active=\\"false\\" divider=\\"true\\" header=\\"false\\"></dropdown-menu-item-stub>
  <dropdown-menu-item-stub active=\\"true\\" divider=\\"false\\" header=\\"false\\" label=\\"Item 4\\"></dropdown-menu-item-stub>
</ul>"
`;
